### ISSUE TITLE
<!DOCTYPE html> gets encoded

### ISSUE THREAD

**Issue by isao** [chjj/marked#354]
HTML is allowed in markdown and marked, but marked@0.3.1 encodes the doctype.

```
bash-3.2$ echo '<!DOCTYPE html>' | marked 
<p>&lt;!DOCTYPE html&gt;</p>

bash-3.2$ echo '<!DOCTYPE html>' | marked --pedantic
<p>&lt;!DOCTYPE html&gt;</p>
```

I imagine this is not a common requirement, but markdown.pl (at least as implemented in BBEdit) passes the doctype un-encoded, as expected.

**Comment by Haeresis** [comment]
If you want kept the doctype in your HTML when you parse an entire HTML page but you want transform others occurences you could use the following code

```js
var marked = require("marked"),
    regex = /^<p>\&lt\;(\!DOCTYPE[^>]*)\&gt\;(.*)<\/p>/i,
    html = "<!DOCTYPE html><html><head>Test</head><body><p><!DOCTYPE html></p></body></html>";

html = marked(html);
html = html.replace(regex, "<$1>$2");

return html;
// <!DOCTYPE html><html><head><title>Test</title></head><body><p>&lt;!DOCTYPE html&gt;</p></body></html>
```

**Comment by aminya** [comment]
Is this issue still a thing?

**Comment by UziTech** [comment]
@aminya I think this was fixed. Are you running into the same issue?

**Comment by aminya** [comment]
I was looking at some code in atom-ide-markdown-service that was applying some workaround for this. I want to know if we need to keep this workaround.
https://github.com/atom-ide-community/atom-ide-markdown-service/blob/b7ff14f2cb6774fb115522f41a4ae3747e9228a7/lib/renderer.js#L109

**Comment by UziTech** [comment]
I don't think the work around is needed anymore. [demo](https://marked.js.org/demo/?outputType=html&text=%3C!DOCTYPE%20html%3E%0A%0A**some%20markdown**&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=master)

It looks like this was fixed in v0.4.0

**Comment by aminya** [comment]
Awesome! I would remove the workaround then.
